infrastructure/persistence/bigquery.py

```
@@ -51,7 +63,30 @@ def execute_query(self, sql_query: str) -> pd.DataFrame:
+                        )
+                        if bytes_processed is not None:
+                            tracer.log_metrics({"dry_run_bytes": bytes_processed})
+                        logger.debug(
+                            "Dry run successful (bytes processed: %s)", bytes_processed
+                        )
+                    except Exception as exc:
+                        logger.error(
+                            "BigQuery dry run failed for query:\n%s\nError: %s",
+                            sql_query,
+                            exc,
+                        )
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)